test/treasure.utils.spec.js

```diff
@@ -22,5 +22,14 @@ describe('Treasure Utils', function () {
       var uuidRegex = /^[A-F0-9]{8}(?:-?[A-F0-9]{4}){3}-?[A-F0-9]{12}$/i
       expect(uuidRegex.test(generateUUID())).to.equal(true)
     })
+    it('generate non colliding UUID', function () {
+      this.timeout(60000)
```

Comment:
I'm not sure that this tests both legs of the ternary.  Maybe you should export both of those functions for testing, and have a `generateUUID` which contains the ternary switch between functions?